cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/TimePartitionedFileSet.java

```diff
@@ -62,25 +63,39 @@
 
   /**
    * Adds a new metadata entry for a particular partition.
-   * Note that existing entries can not be updated.
+   * If the metadata key already exists, it will be overwritten.
    *
    * @param time the partition time in milliseconds since the Epoch
-   *
-   * @throws DataSetException in case an attempt is made to update existing entries.
+   * @throws PartitionNotFoundException when a partition for the given time is not found
```

Comment:
It is changing the semantics, but the behavior is more relaxed now than before, so I don't think thats incompatibility (we're reducing the error cases). Even then, its marked @Beta, so I think that change in the semantics is fair (errors thrown in fewer scenarios). I'll also include a note for this in the release notes.